### PR TITLE
:sparkles: Add clean_content property on UserMessage

### PR DESCRIPTION
**Pull request by beastmatser**
### Changes

-   `adds`: A property on `UserMessage` which removes any markdown syntax

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

**Comment by beastmatser** [comment]
I am not a fan of my own implementation, open for suggestions!

File: pincer/objects/message/user_message.py [hunk 1]
```diff
@@ -539,3 +540,17 @@ async def delete(self):
...
+    @property
+    def clean_content(self):
+        """
+        The message content with any special characters removed.
+        """
+        remove_bold = sub(r"\*\*(.*?)\*\*", r"\1", self.content)
+        remove_italic1 = sub(r"\*(.*?)\*", r"\1", remove_bold)
+        remove_italic2 = sub(r"_(.*?)_", r"\1", remove_italic1)
+        remove_bold_italic = sub(r"\*\*\*(.*?)\*\*\*", r"\1", remove_italic2)
+        remove_underline = sub(r"\_\_(.*?)\_\_", r"\1", remove_bold_italic)
```
Comment (trag1c) [inline pincer/objects/message/user_message.py:553]: hmmmmm
```suggestion
        new_content = self.content
        subs = [
                r"\*\*(.*?)\*\*", # bold
                r"\*(.*?)\*", # italic
                r"_(.*?)_", # italic2
                r"\*\*\*(.*?)\*\*\*", # bold+italic
                r"\_\_(.*?)\_\_" # underline
        ]
        for i in subs:
                new_content = sub(i, r"\1", new_content)
```

File: pincer/objects/message/user_message.py [hunk 2]
```diff
@@ -546,11 +546,16 @@ def clean_content(self):
...
+        subs = [
+                r"\*\*(.*?)\*\*", # bold
+                r"\*(.*?)\*", # italic
+                r"_(.*?)_", # italic2
+                r"\*\*\*(.*?)\*\*\*", # bold+italic
+                r"\_\_(.*?)\_\_" # underline
+        ]
+        for i in subs:
+                new_content = sub(i, r"\1", new_content)
         remove_code_blocks = sub(r"(.*?)```[a-zA-Z]+(\s*)+\n((?:.|\s)*?)```", r"\1\2\3", remove_underline)
```
Comment (Sigmanificient) [inline pincer/objects/message/user_message.py:559]: `remove_underline` => `new_content`

File: pincer/objects/message/user_message.py [hunk 3]
```diff
@@ -546,11 +546,16 @@ def clean_content(self):
         """
         The message content with any special characters removed.
         """
-        remove_bold = sub(r"\*\*(.*?)\*\*", r"\1", self.content)
-        remove_italic1 = sub(r"\*(.*?)\*", r"\1", remove_bold)
-        remove_italic2 = sub(r"_(.*?)_", r"\1", remove_italic1)
-        remove_bold_italic = sub(r"\*\*\*(.*?)\*\*\*", r"\1", remove_italic2)
-        remove_underline = sub(r"\_\_(.*?)\_\_", r"\1", remove_bold_italic)
+        new_content = self.content
+        subs = [
```
Comment (Sigmanificient) [inline pincer/objects/message/user_message.py:550]: i guess can be extract out of the function to a constant

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pincer/objects/message/user_message.py [hunk 4]
```diff
@@ -539,3 +540,22 @@ async def delete(self):
...
+        """
+        The message content with any special characters removed.
+        """
+        new_content = self.content
+        subs = [
+                r"\*\*(.*?)\*\*",  # bold
+                r"\*(.*?)\*",  # italic
+                r"_(.*?)_",  # italic2
+                r"\*\*\*(.*?)\*\*\*",  # bold+italic
+                r"\_\_(.*?)\_\_"  # underline
```
Comment (Sigmanificient) [inline pincer/objects/message/user_message.py:555]: missing regex for spoiler, small code (`this`) & crossed out

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

**Review by Sigmanificient — APPROVED** [review]

File: pincer/objects/message/user_message.py [hunk 5]
```diff
@@ -552,7 +552,10 @@ def clean_content(self):
                 r"\*(.*?)\*",  # italic
                 r"_(.*?)_",  # italic2
                 r"\*\*\*(.*?)\*\*\*",  # bold+italic
-                r"\_\_(.*?)\_\_"  # underline
+                r"\_\_(.*?)\_\_",  # underline
+                r"~~(.*?)~~",  # crossed
+                r"`(.*?)`",  # small code blocks
```
Comment (Sigmanificient) [inline pincer/objects/message/user_message.py:557]: oh wait small code block can also be declared with 2 `

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pincer/objects/message/user_message.py (hunk 3 above)
Comment (beastmatser) [reply pincer/objects/message/user_message.py:550]: Can you explain what you mean by this, I don't understand

**Review by Sigmanificient — APPROVED** [review]

File: pincer/objects/message/user_message.py (hunk 3 above)
Comment (Endercheif) [reply pincer/objects/message/user_message.py:550]: I am assuming that @Sigmanificient is saying that the regex can be a constant in the code, defined at the top outside of the functions/classes.

Comment (Sigmanificient) [reply pincer/objects/message/user_message.py:550]: yep

File: pincer/objects/message/user_message.py [hunk 6]
```diff
@@ -539,3 +540,25 @@ async def delete(self):
         await self._http.delete(
             f"/channels/{self.channel_id}/messages/{self.id}"
         )
+
+    @property
+    def clean_content(self):
+        """
+        The message content with any special characters removed.
```
Comment (ooliver1) [inline pincer/objects/message/user_message.py:547]: ``"""single line docstring"""`` its in the python docstring pep, also wheres the return typehint and docstrings.
docstrings for properties in numpy are:
```py
"""The[Type]: description"""
```

**Review by ooliver1 — CHANGES_REQUESTED** [review]

**Comment by Sigmanificient** [comment]
The last test isn't working @beastmatser could you fix it?

File: pincer/objects/message/user_message.py [hunk 7]
```diff
@@ -567,7 +567,7 @@ def clean_content(self) -> str:
             new_content = re.sub(pattern, r"\1", new_content)
 
         return re.sub(
-            re.compile(r"(.*?)```[a-zA-Z]+(\s*)+\n((?:.|\s)*?)```"),
```
Comment (Sigmanificient) [inline pincer/objects/message/user_message.py]: not sure to understand the change (not strong at regex ;-;) mind make a brief explanation?

Comment (trag1c) [reply pincer/objects/message/user_message.py]: Basically changing from exactly 1 `\n` to at least 1 `\n`

Comment (beastmatser) [reply pincer/objects/message/user_message.py]: You thought I understand regex?

**Pull request closed without merging**